envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotUpdaterTest.kt

```diff
@@ -87,6 +88,28 @@ class SnapshotUpdaterTest {
         Locality.LOCAL, "cluster"
     ).toMultiClusterState()
 
+    @Test
+    fun `should generate allServicesGroup snapshots`() {
+        val cache = MockCache()
+
+        val allServicesGroup = AllServicesGroup(communicationMode = XDS)
+
+        cache.setSnapshot(allServicesGroup, uninitializedSnapshot)
+
+        val updater = snapshotUpdater(
+            cache = cache,
+            properties = SnapshotProperties().apply {
+                incomingPermissions.enabled = true
+            },
+            groups = listOf(allServicesGroup)
+        )
+
+        updater.startWithServices("existingService1", "existingService2")
+
+        hasSnapshot(cache, AllServicesGroup(communicationMode = XDS))
+            .hasOnlyClustersFor("existingService1", "existingService2")
```

Comment:
I adding assertions for timeouts to create integration-like test. @slonka Do you know better place for checks?

Reply:
I pushed these assertions.